x-pack/plugins/security_solution/server/lib/risk_engine/routes/risk_score_preview_route.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger } from '@kbn/core/server';
+import { buildSiemResponse } from '@kbn/lists-plugin/server/routes/utils';
+import { transformError } from '@kbn/securitysolution-es-utils';
+import { DEFAULT_RISK_SCORE_PAGE_SIZE, RISK_SCORE_PREVIEW_URL } from '../../../../common/constants';
+import { riskScoresRequestSchema } from '../../../../common/risk_engine/risk_scoring/risk_scores_request_schema';
+import type { SecuritySolutionPluginRouter } from '../../../types';
+import { buildRouteValidation } from '../../../utils/build_validation/route_validation';
+import { riskScoreService } from '../risk_score_service';
+import { getRiskInputsIndex } from '../helpers';
+
+export const riskScorePreviewRoute = (router: SecuritySolutionPluginRouter, logger: Logger) => {
+  router.post(
+    {
+      path: RISK_SCORE_PREVIEW_URL,
+      validate: { body: buildRouteValidation(riskScoresRequestSchema) },
+      options: {
+        tags: ['access:securitySolution'],
+      },
+    },
+    async (context, request, response) => {
+      const siemResponse = buildSiemResponse(response);
+      const esClient = (await context.core).elasticsearch.client.asCurrentUser;
+      const soClient = (await context.core).savedObjects.client;
+      const siemClient = (await context.securitySolution).getAppClient();
+      const riskScore = riskScoreService({
+        esClient,
+        logger,
+      });
+
+      const {
+        after_keys: userAfterKeys,
+        data_view_id: dataViewId,
+        debug,
+        page_size: userPageSize,
+        identifier_type: identifierType,
+        filter,
+        range: userRange,
+        weights,
+      } = request.body;
+
+      try {
+        const index =
+          (dataViewId &&
+            (await getRiskInputsIndex({
+              dataViewId,
+              logger,
+              soClient,
+            }))) ??
+          siemClient.getAlertsIndex();
+
+        const afterKeys = userAfterKeys ?? {};
+        const range = userRange ?? { start: 'now-15d', end: 'now' };
```

Comment:
But shouldn't be it a responsibility of a client to specify those ranges?
From my perspective I better make the range is required params, and it will be more explicit that have some default which can impact API results without client knowing

Reply:
Sounds great!!, I wonder how the user will access this API spec.

Reply:
@nkhristinin I realize now that I misread your comment, apologies. I think that in this case the endpoint choosing a default range is better than aggregating across all alerts by default; in the UI it won't be possible _not_ to send a range, and if folks are using the preview API then "are my parameters correct" should be the first thing to check if receiving incomplete data. I would agree that there should _not_ be a default range for the calculation API 👍 .

@WafaaNasr I expect that in the near future we'll be able to write some tooling for the docs folks to convert these schema to the necessary format, but in the meantime this is just some pre-work to define the API in a common format.